rust/private/rustc.bzl

```diff
@@ -1845,6 +1854,32 @@ def _get_dirname(file):
     """
     return file.dirname
 
+def _add_per_crate_rustc_flags(ctx, args, crate_info, per_crate_rustc_flags):
```

Comment:
Could you write a unit test for this? The function will probably need to be updated to return a list of args instead populating an args reference but to callers I think this would be clearer.

Reply:
I went ahead and added unit tests for the entire rule.